src/mds/MDLog.cc

```diff
@@ -619,8 +652,6 @@ void MDLog::trim(int m)
     max_ev = events_per_segment + 1;
   }
 
-  submit_mutex.lock();
```

Comment:
As part of the changes in my other comment, I think we should leave this lock to avoid code churn and make @leonid-s-usov 's refactor PR rebase easier.